ClangTidy.cmake

```diff
@@ -2,26 +2,30 @@
 
 # Setup clang-tidy target (enabled if toolchain is clang)
 message(STATUS "Setup target: clang-tidy")
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
```

Comment:
Check apple platform together with compiler

Reply:
And below

Reply:
@malirod, there is no need to check for apple platform because "AppleClang" compiler is available only on macOS.

Changes was reverted back.